pkg/build/generator/generator.go

```diff
@@ -432,6 +432,17 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 	}
 
 	build.Name = getNextBuildName(bc)
+	// Retry getting a non-conflicting name 10 times. If, after 10 tries, there
+	// is still a naming conflict, the process will continue until we try and
```

Comment:
"the process will continue" in my mind meant "keep trying more names" ... maybe there's better wording for that.